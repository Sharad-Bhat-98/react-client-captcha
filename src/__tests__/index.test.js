import React from 'react'
import { fireEvent, render, screen } from '@testing-library/react'

import ClientCaptcha from '../index'

jest.fn('../helpers', () => ({
  generateCanvas: jest.fn(),
  generateCode: jest.fn()
}))

const defaultProps = {
  captchaCode: jest.fn()
}

describe('ClientCaptcha', () => {
  it('Should have a default export', () => {
    expect(typeof ClientCaptcha).toBe('object')
  })

  describe('Render', () => {
    it('Should render the ClientCaptcha', () => {
      const { container } = render(<ClientCaptcha {...defaultProps} />)

      expect(container).toMatchSnapshot()
    })
  })

  describe('refresh button', () => {
    it('Should call captchaCode function on refresh', () => {
      render(<ClientCaptcha {...defaultProps} />)
      fireEvent.click(screen.getByTestId('refreshButton'))
      expect(defaultProps.captchaCode).toHaveBeenCalled()
    })

    it('Should not be shown when props if false', () => {
      render(<ClientCaptcha {...defaultProps} refreshButton={false} />)
      expect(screen.queryByTestId('refreshButton')).toBeNull()
    })
  })
})
