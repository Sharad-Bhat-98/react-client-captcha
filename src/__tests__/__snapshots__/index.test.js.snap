// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientCaptcha Render Should render the ClientCaptcha 1`] = `
<div>
  <div
    class="captchaContainer"
  >
    <canvas
      class=""
      height="40"
      style="pointer-events: none;"
      width="100"
    />
    <button
      class="retryButton"
      data-testid="refreshButton"
    >
      <img
        alt="Re-new captcha"
        class=""
        height="24"
        src="https://cdn.jsdelivr.net/npm/react-client-captcha/dist/retry.svg"
        width="24"
      />
    </button>
  </div>
</div>
`;
